crates/storage/src/collections/unordered_set.rs

```diff
@@ -0,0 +1,267 @@
+use sha2::{Digest, Sha256};
+
+// fixme! macro expects `calimero_storage` to be in deps
+use crate as calimero_storage;
+use crate::address::{Id, Path};
+use crate::collections::error::StoreError;
+use crate::entities::{Data, Element};
+use crate::interface::Interface;
+use crate::{AtomicUnit, Collection};
+
+/// A set collection that stores unqiue values once.
+#[derive(AtomicUnit, Clone, Debug, Eq, PartialEq, PartialOrd)]
+#[type_id(253)]
+#[root]
+pub struct UnorderedSet<V> {
+    /// The prefix used for the set's entries.
+    id: Id,
+    /// The entries in the set.
+    entries: Entries<V>,
+    /// The storage element for the set.
+    #[storage]
+    storage: Element,
+}
+
+/// A collection of entries in a set.
+#[derive(Collection, Clone, Debug, Eq, PartialEq, PartialOrd)]
+#[children(Entry<V>)]
+struct Entries<V> {
+    /// Helper to associate the generic types with the collection.
+    _priv: PhantomData<V>,
+}
+
+/// An entry in a set.
+#[derive(AtomicUnit, Clone, Debug, Eq, PartialEq, PartialOrd)]
+#[type_id(252)]
+pub struct Entry<V> {
+    /// The value for the entry.
+    value: V,
+    /// The storage element for the entry.
+    #[storage]
+    storage: Element,
+}
+
+impl<V: BorshSerialize + BorshDeserialize + AsRef<[u8]> + PartialEq> UnorderedSet<V> {
+    /// Create a new set collection.
+    ///
+    /// # Errors
+    ///
+    /// If an error occurs when interacting with the storage system, or a child
+    /// [`Element`](crate::entities::Element) cannot be found, an error will be
+    /// returned.
+    ///
+    pub fn new() -> Result<Self, StoreError> {
+        let id = Id::random();
+        let mut this = Self {
+            id: id,
+            entries: Entries::default(),
+            storage: Element::new(&Path::new(format!("::unused::set::{id}::path"))?, Some(id)),
+        };
+
+        let _ = Interface::save(&mut this)?;
+
+        Ok(this)
+    }
+
+    /// Compute the ID for a value in the set.
+    fn compute_id(&self, value: &[u8]) -> Id {
+        let mut hasher = Sha256::new();
+        hasher.update(self.id.as_bytes());
+        hasher.update(value);
+        Id::new(hasher.finalize().into())
+    }
+
+    /// Insert a value pair into the set collection if the element does not already exist.
+    ///
+    /// # Errors
+    ///
+    /// If an error occurs when interacting with the storage system, or a child
+    /// [`Element`](crate::entities::Element) cannot be found, an error will be
+    /// returned.
+    ///
+    pub fn insert(&mut self, value: V) -> Result<bool, StoreError> {
+        let path = self.path();
+
+        if self.contains(&value)? {
+            return Ok(false);
+        } else {
+            let storage = Element::new(&path, Some(self.compute_id(value.as_ref())));
+            let _ = Interface::add_child_to(
+                self.storage.id(),
+                &mut self.entries,
+                &mut Entry { value, storage },
+            )?;
+        }
```

Comment:
early return helps save on indentation

```suggestion
            return Ok(false);
        }

        let storage = Element::new(&path, Some(self.compute_id(value.as_ref())));
        let _ = Interface::add_child_to(
            self.storage.id(),
            &mut self.entries,
            &mut Entry { value, storage },
        )?;
```